backend/routes/UserRoutes.js

```diff
@@ -64,7 +64,9 @@ router.get("/:id", (req, res, next) => {
         // checks if a user was found with id
         res.status(404).send("No user found with provided id");
       } else {
-        res.status(200).json(user);
+        const currentUserId = idOfCurrentUser(req);
```

Comment:
Why can't we just do this?

```suggestion
        const currentUserId = req.user._id;
```

The user's token has already been validated by the passport middleware by this point, so req.user will always be populated -- parsing the auth header on our own does the same thing that passport does.

Reply:
Looking over the rest of the file the `idOfCurrentUser` function was added/used by an earlier PR, we should fix that